mantidproject/mantid
# Add steps to test main to correctly init gmock

**Description of work.**
Currently we do not init GMock correctly, as per
https://chromium.googlesource.com/external/github.com/google/googletest/+/refs/tags/release-1.8.0/googlemock/docs/ForDummies.md

This adds the missing steps, and removes a file which does not partake
in any sort of main generation.

By adding this ctest will correctly mark GMock expectation failures as
failures instead of passing if TS_ASSERT(VERIFY_AND_CLEAR(x)) pattern is
not used or missed

**To test:**
- Check tests pass
- Change an EXPECT_CALL in a random test to an incorrect value (that does not have an explicit verify) you should see CTest correctly fail it now

Part of #31069 





---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
